src/zenml/artifacts/artifact_config.py

```diff
@@ -83,3 +91,35 @@ def _remove_old_attributes(cls, data: Dict[str, Any]) -> Dict[str, Any]:
             )
 
         return data
+
+    @model_validator(mode="after")
+    def artifact_config_after_validator(self) -> "ArtifactConfig":
```

Comment:
I'm wondering why this is happening as an after validator. This means if I run the same pipeline twice in one python script, it will evaluate just once and therefore reuse the same name?

If that is indeed the case, is this intended? IMO I'd expect the callable to be evaluated each time a pipeline is being run, and we could easily do that by just doing the evaluation in the `_evaluated_name` property?

Reply:
This is a valid concern, I acknowledge it. I started exploring this and it turned out to be quite a complex issue to solve (though I would still do it). It hits exactly the same limitations with step configs being saved and reused from the deployment, so I would need to implement something to break this circle here already, not you as part of Lazy Loaders rework.